store/catalog.go

```diff
@@ -34,24 +34,24 @@ func NewCatalog(databaseID *int, store *Store, dbType db.Type) *Catalog {
 }
 
 // GetDatabase implements the catalog.Catalog interface.
-func (c *Catalog) GetDatabase(ctx context.Context) (*catalog.Database, error) {
+func (c *Catalog) GetDatabase(ctx context.Context) (*catalog.Database, *catalog.Context, error) {
 	if c.databaseID == nil {
```

Comment:
Not directly related to this PR, but I am wondering why we make databaseID a pointer. I check all existing caller would pass a valid databaseID. https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/store/catalog.go?L28-34&subtree=true

Also, from the code perspective, the store/catalog.go should always have the catalog info. It's the caller's responsibility to either create a store.Catalog or an EmptyCatalog.

Reply:
FYI https://github.com/bytebase/bytebase/pull/2613